File: src/main/java/it/unical/demacs/enterprise/fintedapp/dto/UserProfileDto.java
package it.unical.demacs.enterprise.fintedapp.dto;

import java.util.Date;
import java.util.List;

import jakarta.validation.constraints.Email;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.ToString;

@Data
@NoArgsConstructor
@SuppressWarnings(value = { "unused" })
@ToString
public class UserProfileDto {
	
	private Long id;
	
	private String firstName;
	
	private String lastName;
	
	private String username;
	
	private Date registrationDate;
	
	@Email
	private String credentialsEmail;
	
	private String addressRoute;
	
	private String addressNumber;
	
	private String addressCity;
	
    private List<PostDto> publishedPosts;

    private List<ReviewDto> receivedReviews;
}

File: FINTed_android/app/src/main/java/it/unical/demacs/enterprise/fintedapp/models/UserProfileDto.kt
package it.unical.demacs.enterprise.fintedapp.models

/**
 *
 * @param id
 * @param firstName
 * @param lastName
 * @param username
 * @param registrationDate
 * @param credentialsEmail
 * @param addressRoute
 * @param addressNumber
 * @param addressCity
 * @param publishedPosts
 * @param receivedReviews
 */
data class UserProfileDto (

    val id: kotlin.Long? = null,
    val firstName: kotlin.String? = null,
    val lastName: kotlin.String? = null,
    val username: kotlin.String? = null,
    val registrationDate: java.util.Date? = null,
    val credentialsEmail: kotlin.String? = null,
    val addressRoute: kotlin.String? = null,
    val addressNumber: kotlin.String? = null,
    val addressCity: kotlin.String? = null,
    val publishedPosts: kotlin.Array<PostDto>? = null,
    val receivedReviews: kotlin.Array<ReviewDto>? = null
) {
}
